File: Ticket_Raising_Backend/TicketClassLibrary/Models/Employee.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
 
namespace Helpdesk.Models
{
    public class Employee
    {
        [Key]
        public int EmpId { get; set; }
 
        [Required]
        public string FirstName { get; set; }
 
        public string LastName { get; set; }
 
        [Required]
        public string Email { get; set; }
 
        [Required]
        public string Password { get; set; }
 
        public string Role { get; set; }   
 
        public string PhoneNumber { get; set; }
 
        public ICollection<Ticket> Tickets { get; set; }
        public ICollection<TicketComment> TicketComments { get; set; }
        public ICollection<TicketAssignment> TicketAssignments { get; set; }
    }
}

File: Ticket_Raising_Frontend/Ticket_Raising_App/src/Models/Employee.ts
export class Employee {
 
    public empId: string;

    public firstName: string;

    public lastName: string;

    public email: string;

    public password: string;

    public role: string;

    public phoneNumber: string;
 
    constructor(

        empId: string = "",

        firstName: string = "",

        lastName: string = "",

        email: string = "",

        password: string = "",

        role: string = "",

        phoneNumber: string = ""

    ) {

        this.empId = empId;

        this.firstName = firstName;

        this.lastName = lastName;

        this.email = email;

        this.password = password;

        this.role = role;

        this.phoneNumber = phoneNumber;

    }

}

 